keras_retinanet/utils/image.py

```diff
@@ -195,3 +195,170 @@ def resize_image(img, min_side=800, max_side=1333):
     img = cv2.resize(img, None, fx=scale, fy=scale)
 
     return img, scale
+
+
+def _uniform(val_range):
+    """
+    Uniformly sample from the given range
+
+    Args
+        val_range: A pair of lower and upper bound
+    """
+    return np.random.uniform(val_range[0], val_range[1])
+
+
+def _check_range(val_range, min_val=None, max_val=None):
+    """
+    Check whether the range is a valid range
+
+    Args
+        val_range: A pair of lower and upper bound
+        min_val: minimal lower bound
+        max_val: maximum upper bound
+    """
+    if val_range[0] > val_range[1]:
+        raise ValueError('interval lower bound > upper bound')
+    if min_val is not None and val_range[0] < min_val:
+        raise ValueError('invalid interval lower bound')
+    if max_val is not None and val_range[1] > max_val:
+        raise ValueError('invalid interval upper bound')
+
+
+def _clip(image):
+    """
+    Clip and convert an image to np.uint8
+
+    Args
+        image: Image to clip
+    """
+    return np.clip(image, 0, 255).astype(np.uint8)
+
+
+class VisualEffect:
+    """ Struct holding parameters for applying a visual effect.
+
+    Args
+        contrast_factor:   contrast factor between 0 and 3
```

Comment:
Could you modify the documentation to be sentences (start with uppercase and end with period). Also, if possible in a few words, try to explain how the values are used to modify the image.

Reply:
I tried it with tensorflow functions following the examples in the `backend` module. However, I had to modify/replace opencv and numpy code in several files. And it didn't work after that anyway. I would have to make more changes I suppose.

I would prefer keeping the current numpy/opencv implementation as it works well with `debug.py` and it would be easier to extend.

Reply:
> However, I had to modify/replace opencv and numpy code in several files.

Hmm what changes did you have to make? I would think it just works out of the box.

Reply:
OpenCV functions  (eg., in apply_transform, resize_image which follows visual effect) works with numpy arrays, but tf.image functions return Tensors. So I'd have to replace any numpy/opencv code to tensorflow.